library/ns1_zone.py

```diff
@@ -243,39 +311,50 @@ def update(self, zone):
 
         self.module.exit_json(changed=changed, id=zone['id'], data=zone.data)
 
+    def create(self, zone):
+        if self.module.check_mode:
+            # short circuit in check mode
```

Comment:
wish we didn't need these comments but i can't think of an alternative that isn't clunkier in other ways. so, glad you put 'em there

Reply:
Still refactoring program flow to get to the point where there is only one happy path through the module.  Think we can remove this once the module only exits in a single place.